### ISSUE TITLE
simultaneously training two twin models with the same data and getting different results

### ISSUE THREAD

**Issue by yotamin** [keras-team/keras#10741]
I'm trying to train the same model at the same time as such:

import numpy as np
import os
import tensorflow as tf
import random as rn
os.environ['PYTHONHASHSEED'] = '0'
seed = 3
np.random.seed(seed)
rn.seed(12345)
session_conf = tf.ConfigProto(intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
from keras import backend as K
tf.set_random_seed(1234)
sess = tf.Session(graph=tf.get_default_graph(), config=session_conf)
K.set_session(sess)

model1 = load_model('my_model.h5')
model1..compile(loss='mse', optimizer='adam')
model2 = load_model('my_model.h5')
model2..compile(loss='mse', optimizer='adam')

for batch in batches:
   model1.train_on_batch(batch[0], batch[1])
   model2.train_on_batch(batch[0], batch[1])

When I'm evaluating against the same test set I get different results for each model, no idea why.
